src/components/FileStorageService.js

```diff
@@ -11,19 +11,29 @@ goog.provide('ga_filestorage_service');
       var fileStorageUrl = this.fileStorageUrl;
       var publicUrl = this.publicUrl;
 
-      var getServiceUrl = function(id) {
-        return fileStorageUrl + ((id) ? '/' + id : '');
+      // custom: This param override the default 'files' endpoint. To 'glstyles'
+      //         for example.
+      var getServiceUrl = function(id, custom) {
+        var url = fileStorageUrl;
+        if (custom) {
+          url = url.replace(/files$/, custom);
+        }
+        return url + ((id) ? '/' + id : '');
       };
 
-      var getPublicUrl = function(fileId) {
-        return publicUrl + ((fileId) ? '/' + fileId : '');
+      var getPublicUrl = function(fileId, custom) {
```

Comment:
I would pass custom once in the constructor of `FileStorage` and make these methods private so that you have access to the basePath.

Reply:
As you said on friday I can manage depending on content type and as I said using a 'gl-styles/dfgfgdfg'  as fileId.

Reply:
I will do 2 services is more clean